docs/integrating-uptest-for-e2e-testing.md

```diff
@@ -0,0 +1,143 @@
+	\${KUBECTL} -n upbound-system wait --for=condition=Available deployment --all --timeout=5m
+
+	echo "Creating a default provider config..."
+	cat <<EOF | \${KUBECTL} apply -f -
+	apiVersion: github.upbound.io/v1beta1
+	kind: ProviderConfig
+	metadata:
+	  name: default
+	spec:
+	  credentials:
+	    source: Secret
+	    secretRef:
+	      name: provider-secret
+	      namespace: upbound-system
+	      key: credentials
+	EOF
+	```
+
+5. Now, please add the following lines to your Makefile which which will add `uptest` and `e2e` targets:
+
+	```Makefile
+	# ====================================================================================
+	# End to End Testing
+	CROSSPLANE_NAMESPACE = upbound-system
+	-include build/makelib/local.xpkg.mk
+	-include build/makelib/controlplane.mk
+
+	uptest: $(UPTEST) $(KUBECTL) $(KUTTL)
+		@$(INFO) running automated tests
+		@KUBECTL=$(KUBECTL) KUTTL=$(KUTTL) $(UPTEST) e2e "${UPTEST_EXAMPLE_LIST}" --setup-script=cluster/test/setup.sh || $(FAIL)
+		@$(OK) running automated tests
+
+	e2e: build controlplane.up local.xpkg.deploy.provider.$(PROJECT_NAME) uptest
+	```
+
+## Testing Locally
+
+1. Generate a [Personal Access Token](https://github.com/settings/tokens/new) for your Github account with
+    `repo/public_repo` and `delete_repo` scopes.
+2. Run the following:
+
+	```bash
+	export UPTEST_CLOUD_CREDENTIALS=<your-token-here>
+	UPTEST_EXAMPLE_LIST=examples/repository/repository.yaml make e2e
+	```
+
+You should see a `PASS` at the end of logs indicating everything worked fine.
+
+## Adding the GitHub workflow
+
+Now we have things working locally, let's add a GitHub workflow to automate end to end testing with CI.
+
+1. Run the following to add the GitHub workflow definition which will be triggered for `issue_comment` events and will call
+uptests reusable workflow:
+
+	```bash
+	cat <<EOF > .github/workflows/e2e.yaml
+	name: End to End Testing
+
+	on:
+	  issue_comment:
+	    types: [created]
+
+	jobs:
+	  e2e:
+	    uses: upbound/uptest/.github/workflows/pr-comment-trigger.yml@main
+	    secrets:
+	      UPTEST_CLOUD_CREDENTIALS: \${{ secrets.UPTEST_CLOUD_CREDENTIALS }}
+	      UPTEST_DATASOURCE: \${{ secrets.UPTEST_DATASOURCE }}
+	EOF
+	```
+	
+2. Commit and push to the `main` branch of the repository.
+
+	```
+	git add .github/workflows/e2e.yaml
+	git commit -s -m "Add e2e workflow"
+	git push origin main
+	```
+
+3. Lastly, we need to add a Repository Secret with our GitHub token.
+	1. Go to your repository settings in GitHub UI.
+	2. On the left side, select `Secrets` -> `Actions` under `Security` section.
+	3. Hit `New Repository Secret`
+	4. Enter `UPTEST_CLOUD_CREDENTIALS` as `Name` and your GitHub Token as `Secret` and hit `Add secret`
+
+## Testing via CI
+
+We are now ready to test our changes end to end via GitHub Actions. We will try that out by opening a test PR.
+
+1. Go to the `examples/repository/repository.yaml` file and make some wording changes in `description` field, e.g. 
```

Comment:
Strictly following the guide, just before the sample file modification, the git status of the repo is not clean:

```
git status --short
 M Makefile
?? cluster/test/
?? kubeconfig
```
So we need to
* Instruct the reader to commit `Makefile` and `cluster/test/setup.sh` somewhere before this step.
* Add `kubconfig` to `.gitignore` of https://github.com/upbound/demo-uptest-integration/blob/main/.gitignore

Reply:
Both done!